src/client/sandbox/code-instrumentation/location/ancestor-origins-wrapper.js

```diff
@@ -0,0 +1,61 @@
+import nativeMethods from '../../native-methods';
+import LocationAccessorsInstrumentation from './index';
+
+export default function DOMStringListWrapper (window, getCrossDomainOrigin) {
+    const nativeOrigins = window.location.ancestorOrigins;
+    let parentWindow    = window;
+
+    this._nativeLength = nativeOrigins.length;
+
+    for (let i = 0; i < this._nativeLength; i++) {
+        parentWindow                = parentWindow.parent;
+        const parentLocationWrapper = LocationAccessorsInstrumentation.getLocationWrapper(parentWindow);
+        const isCrossDomainParent   = parentLocationWrapper === parentWindow.location;
+
+        // eslint-disable-next-line no-restricted-properties
+        updateOrigin(this, i, isCrossDomainParent ? '' : parentLocationWrapper.origin);
+
+        if (isCrossDomainParent) {
+            getCrossDomainOrigin(parentWindow, origin => {
+                updateOrigin(this, i, origin);
+            });
+        }
+    }
+}
+
+DOMStringListWrapper.prototype = nativeMethods.objectCreate.call(Object, DOMStringList.prototype);
```

Comment:
Let's try to use:
```js
DOMStringListWrapper.prototype = nativeMethods.objectCreate(DOMStringList.prototype);
```

Reply:
And there https://github.com/DevExpress/testcafe-hammerhead/pull/1804/files#diff-d04bb877945ffff88eb1829bf6627b47R44